tests/completion_check.rs

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Test the completion command.
+
+use assert_cmd::Command;
+
+/// This test checks the CLI interface for the completion command.
+#[test]
+fn test_generate_completion() {
+    let mut cmd = Command::cargo_bin("weaver").unwrap();
+    let output = cmd
+        .arg("completion")
+        .arg("bash")
+        .timeout(std::time::Duration::from_secs(60))
+        .output()
+        .expect("failed to execute process");
+    assert!(output.status.success());
+    let stdout = String::from_utf8(output.stdout).expect("Invalid UTF-8");
+    assert!(
+        stdout.contains("weaver__completion"),
```

Comment:
To really regression test this well we would need to compare the expected output of `generate_completion` with what was captured on `stdout` and they should be identical.

To do this we would need to add an option to `weaver completion` like `--completion-path` where `generate_completion` would write to directly.

Then in your test you would run `weaver completion --completion-path tempfile.txt`, then read back in `tempfile.txt`, then run your test capturing stdout and compare.

The problem is that weaver uses stdout for other diagnostic information and we have problems separating the artifact delivered to stdout from the diagnostics. See https://github.com/open-telemetry/weaver/issues/184#issuecomment-2781465337

Reply:
Sorry that's not what I meant. We're trying to guard against weaver accidentally sending any text to stdout before or after the completion text. If it does then the completion script is broken.

My idea was to have an optional file path config item, but KEEP the default stdout behaviour. You then run the weaver command TWICE. The first time to capture the expected output by using the file path option. Then the second time to capture the observed output using stdout. Then check expected == observed.

Reply:
Ahhh... That makes sense. I'll it updated.